code/modules/mob/living/simple_animal/bot/bot.dm

```diff
@@ -471,21 +484,23 @@ Pass a positive integer as an argument to override a bot's default speed.
 	var/area/end_area = get_area(waypoint)
 
 	if(client) //Player bots instead get a location command from the AI
-		to_chat(src, "<span class='noticebig'>Priority waypoint set by [icon2html(caller, src)] <b>[caller]</b>. Proceed to <b>[end_area.name]<\b>.")
+		to_chat(src, "<span class='noticebig'>Priority waypoint set by [icon2html(caller, src)] <b>[caller]</b>. Proceed to <b>[end_area.name]<\b>.</span>")
 
 	//For giving the bot temporary all-access.
 	var/obj/item/card/id/all_access = new /obj/item/card/id
 	var/datum/job/captain/All = new/datum/job/captain
 	all_access.access = All.get_access()
 
-	path = get_path_to(src, waypoint, /turf/proc/Distance_cardinal, 0, 200, id=all_access)
+	set_path(get_path_to(src, waypoint, /turf/proc/Distance_cardinal, 0, 200, id=all_access))
 	calling_ai = caller //Link the AI to the bot!
 	ai_waypoint = waypoint
 
 	if(path && path.len) //Ensures that a valid path is calculated!
 		if(!on)
 			turn_on() //Saves the AI the hassle of having to activate a bot manually.
 		access_card = all_access //Give the bot all-access while under the AI's command.
+		if(client)
+			addtimer(CALLBACK (src, .proc/bot_reset), 300, TIMER_UNIQUE) //if the bot is player controlled, they only get the extra access for 30 seconds.

```

Comment:
Two recommendations: Extend the access timer to 60 seconds; 30 is not always enough to get from one section of Box/Meta to another.
Second, consider adding a message to the player here informing him/her about the all-access timer. Otherwise, only those having read the contents of this PR would be aware of this mechanic.